src/main/java/mate/academy/dao/impl/OrderDaoImpl.java

```diff
@@ -0,0 +1,54 @@
+package mate.academy.dao.impl;
+
+import java.util.List;
+import mate.academy.dao.OrderDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Order;
+import mate.academy.model.User;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class OrderDaoImpl implements OrderDao {
+    @Override
+    public List<Order> getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<Order> query = session.createQuery("FROM Order o "
+                    + "LEFT OUTER JOIN FETCH o.tickets t "
+                    + "LEFT OUTER JOIN FETCH t.movieSession ms "
+                    + "LEFT OUTER JOIN FETCH ms.cinemaHall "
+                    + "LEFT OUTER JOIN FETCH ms.movie "
+                    + "WHERE o.user.id = :id", Order.class);
+            query.setParameter("id", user.getId());
+            return query.getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can`t get orders list for user: "
+                    + user + " by user id: " + user.getId(), e);
+        }
+    }
+
+    @Override
+    public Order add(Order order) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(order);
+            transaction.commit();
+            return order;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add order: " + order, e);
```

Comment:
Similar to the previous comment, providing more context in the error message can be beneficial. Consider including details about the order that failed to be added.